test/testenv/s3utils.go

```diff
@@ -226,3 +229,74 @@ func UploadFilesToS3(testS3Bucket string, s3TestDir string, applist []string, do
 	}
 	return uploadedFiles, nil
 }
+
+// DisableAppsToS3 download apps from S3, untar them, modify their conf file to disable them, re-tar and upload the disabled version to S3
+func DisableAppsToS3(testDataS3Bucket string, s3AppDir string, downloadDir string, appFileList []string, s3TestDir string) ([]string, error) {
+
+	// Download apps tar files
+	DownloadFilesFromS3(testDataS3Bucket, s3AppDir, downloadDir, appFileList)
+
+	// Create a folder named 'untarred_apps' to store untarred apps folders
+	untarredAppsMainFolder := downloadDir + "/untarred_apps"
+	cmd := exec.Command("mkdir", untarredAppsMainFolder)
+	cmd.Run()
+
+	// Create a folder named 'disabled_apps' to stored disabled apps tgz files
+	disabledAppsFolder := downloadDir + "/disabled_apps"
+	cmd = exec.Command("mkdir", disabledAppsFolder)
+	cmd.Run()
+
+	for _, key := range appFileList {
+		// Create a specific folder for each app in 'untarred_apps'
+		tarfile := downloadDir + "/" + key
+		lastInd := strings.LastIndex(key, ".")
+		untarredCurrentAppFolder := untarredAppsMainFolder + "/" + key[:lastInd]
+		cmd := exec.Command("mkdir", untarredCurrentAppFolder)
+		cmd.Run()
+
+		// Untar the app
+		cmd = exec.Command("tar", "-xf", tarfile, "-C", untarredCurrentAppFolder)
+		cmd.Run()
+
+		// Disable the app
+		// - Get the name of the untarred app folder (as it could be different from the tgz file)
+		wildcardpath := untarredCurrentAppFolder + "/*/./"
+		bytepath, _ := exec.Command("/bin/sh", "-c", "cd "+wildcardpath+"; pwd").Output()
+		untarredAppRootFolder := string(bytepath)
+		untarredAppRootFolder = untarredAppRootFolder[:len(untarredAppRootFolder)-1] //removing \n at the end of folder path
+
+		// - Edit /default/app.conf (add "state = disabled" in [install] stanza)
+		appConfFile := untarredAppRootFolder + "/default/app.conf"
+		input, err := ioutil.ReadFile(appConfFile)
+		if err != nil {
+			log.Fatalln(err)
+			return nil, err
+		}
+		lines := strings.Split(string(input), "\n")
+		for i, line := range lines {
+			if strings.Contains(line, "[install]") {
+				lines[i] = "[install]\nstate = disabled"
+			}
+			if strings.Contains(line, "state = enabled") {
+				lines = append(lines[:i], lines[i+1:]...)
+			}
```

Comment:
Sorry did not understand the above lines. If we are already adding `disabled` in lines 277-279 then what does lines 280-282 does?

Reply:
Some apps I used for testing were having "state = enabled" inside the conf file, so it needs to be removed to avoid having both enabled and disabled at the same time in the file.